// @models/Content.js
const dynamoose = require('dynamoose');

/*"hashtags": ["dance", "yoga"],
    "fileLocation": "https://gazettereview.com/wp-content/uploads/2015/05/cat.jpg",
    "location": 
      {
        "city": "san francisco",
        "state": "ca"
      },
    "uploadDate": "10/01/18",
    "uploadTime": "12:00:00pm",
    "likes": "42",
    "shares": "10",
    "comments": 
      {
        "user1": "really fun event!",
        "user2": "cool =D"
      }
      */
const Schema = dynamoose.Schema;
var ContentSchema = new Schema({
  content_uuid: {
    type: String,
    hashKey: true
  },
  hashtags: {
    type: Array
  },
  title: {
    type: String,
    require: 'true'
  },
  fileLocation: {
    type: String
  },
  video_id: {
    type: String
  },
  location: {
    type: Object 
  },
  biz_uuid: {
    type: String,
    index: {
      global: true,
      rangeKey: 'content_uuid',
      name: 'biz_uuid_index',
      project: true, // ProjectionType: ALL
      throughput: 5 // read and write are both 5
    }
  },
  user_uuid: {
    type: String,
    index: {
      global: true,
      rangeKey: 'content_uuid',
      name: 'user_uuid_index',
      project: true, // ProjectionType: ALL
      throughput: 5 // read and write are both 5
    }
  },
  updateDate: {
    type: String
  },
  uploadTime: {
    type: String
  },
  likes: {
    type: String
  },
  shares: {
    type: String
  },
  comments: {
    type: Object
  }
}, {
  throughput: {
    read: 5,
    write: 2
  },
  timestamps: {
    createdAt: 'creationDate',
    updatedAt: 'lastUpdateDate'
  }
});


module.exports = {
  schema: ContentSchema
}
